apps/api/src/app/widgets/usecases/get-notifications-feed/get-notifications-feed.usecase.ts

```diff
@@ -50,6 +50,12 @@ export class GetNotificationsFeed {
       });
     }
 
+    for (const message of feed) {
+      if (message._actorId && message.actor?.type === ActorTypeEnum.USER) {
+        message.actor.data = await this.processAvatar(message._actorId, command.environmentId);
+      }
+    }
```

Comment:
Unfortunately, this approach is not very performant is we don't utilize parallelization here, nor using caching. 

Could you please this logic to use "populate" from mongoose on the dal level instead of looping here on this?